tasks/compile-example.js

```diff
@@ -84,12 +84,57 @@ module.exports = function(config, updateTimestamp) {
       examples = [];
     }
 
-    // compile example
+    // parse examples into documents and add metadata
     if (file.isBuffer()) {
-      const contents = file.contents.toString();
-      const document = DocumentParser.parse(contents);
-      const stream = this;
       const example = ExampleFile.fromPath(file.path);
+      const contents = file.contents.toString();
+      example.document = DocumentParser.parse(contents);
+      example.file = file;
+      examples.push(example);
+    }
+
+    cb();
+  }
+
+  function endStream(cb) {
+    // no examples passed in, no file goes out
+    if (!latestFile || !examples) {
+      cb();
+      return;
+    }
+
+    const categories = mapToCategories(examples);
```

Comment:
Can you extract a method `compileIndex` as well so that the order is:

```
const stream = this;
const categories = mapToCategories(examples);
compileExamples(examples, categories, stream);
compileIndex(categories, stream);
```